modules/web/src/app/shared/components/application-list/component.ts

```diff
@@ -278,64 +284,91 @@ export class ApplicationListComponent implements OnInit, OnDestroy {
-        let icon = StatusIcon.Pending;
-        let message = '';
-        if (application.deletionTimestamp) {
-          icon = StatusIcon.Error;
-          message = 'Deleting';
+    const loadingAppDefDetails: Record<string, boolean> = {};
+    // load application definition details
+    forkJoin(
+      this.applications.map(application => {
+        const applicationRef = application.spec.applicationRef;
+        if (
+          !application.status &&
+          applicationRef &&
+          !this.applicationDefinitionsMap.get(applicationRef.name)?.spec?.versions &&
+          !loadingAppDefDetails[applicationRef.name]
+        ) {
+          loadingAppDefDetails[applicationRef.name] = true;
+          return this._applicationService.getApplicationDefinition(applicationRef.name).pipe(
+            map(appDef => {
+              this._updateApplicationDefinition(appDef);
+              return application;
+            })
+          );
         }
-        if (status.conditions?.length) {
-          const failingCondition = status.conditions.find(condition => condition.status === 'False');
-          if (failingCondition) {
-            icon = StatusIcon.Error;
-            const error = failingCondition.message;
-            message = `${error} ${
-              error || !error.endsWith('.') ? '.' : ''
-            } Please check your configuration or contact your KKP Administrator.`;
-          } else {
-            icon = StatusIcon.Running;
-            message = 'Ready';
+        return of(application);
+      })
+    )
+      .pipe(take(1))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(_ => {
+        this.applicationsMethodMap = {};
+        this.applicationsSourceMap = {};
+        this.applicationsStatusMap = {};
+        this.applications.forEach(application => {
+          const applicationRef = application.spec.applicationRef;
+          const status = application.status;
+          const namespace = application.spec.namespace.name;
+
+          this.applicationsMethodMap = {
+            ...this.applicationsMethodMap,
+            [namespace]: {
+              ...(this.applicationsMethodMap[namespace] || {}),
+              [application.name]:
+                status?.method || this.applicationDefinitionsMap.get(applicationRef?.name)?.spec.method,
+            },
+          };
+
+          const versionSources =
+            status?.applicationVersion?.template?.source ||
+            this.applicationDefinitionsMap
+              .get(applicationRef?.name)
+              ?.spec.versions?.find(version => version.version === applicationRef?.version)?.template?.source;
+
+          this.applicationsSourceMap = {
+            ...this.applicationsSourceMap,
+            [namespace]: {
+              ...(this.applicationsSourceMap[namespace] || {}),
+              [application.name]: versionSources ? Object.keys(versionSources).find(key => !!versionSources[key]) : '',
+            },
+          };
+          if (status) {
```

Comment:
When is the case when it will be valid? Deletion of application?
currently this code is not executing due to status `undefined`

Reply:
`status` is present when we load applications for a cluster (i.e. on cluster details page).